src/test/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/adjourncase/AdjournCaseAboutToSubmitHandlerTest.java

```diff
@@ -63,53 +65,49 @@ public class AdjournCaseAboutToSubmitHandlerTest {
     private PreviewDocumentService previewDocumentService;
 
     private SscsCaseData sscsCaseData;
-    private AutoCloseable autoCloseable;
+
+    private void setUpAdjournCaseMocks() {
+        when(callback.getEvent()).thenReturn(EventType.ADJOURN_CASE);
+        when(callback.getCaseDetails()).thenReturn(caseDetails);
+        when(caseDetails.getCaseData()).thenReturn(sscsCaseData);
+    }
```

Comment:
Don't do this, either do them in the before each or on each test. a test should be easy to follow and have three parts and when you extract these parts into multiple different methods, it becomes harder to follow and harder to maintain due to tight coupling between tests:
https://www.geeksforgeeks.org/coupling-in-java/
A unit test should be, as per the name, a unit, a single thing tested and when that unit changes you should theoretically only need to change one unit test to match.
They should follow this format so that looking at that test it can be understood:

Given (The test inputs)

When (The test)

Then (The test conditions)

For things that are universal to the class itself we use the @BeforeEach and is the acceptable way to do generics when doing unit tests.

Reply:
Only reason I made this change was because I updated Mockito so then there were unnecessary stubbings in the tests which did not require these when's in the (now) BeforeEach. Extracting to a helper method I thought was a good thing to avoid extensive repetition throughout the tests.

Reply:
resolved with Nested classes

Reply:
Replaced Nested classes with extracting separate test classes which extend a common abstract setup class